framework/src/base/FEProblem.C

```diff
@@ -704,16 +709,47 @@ FEProblem::getMaxScalarOrder() const
 }
 
 void
+FEProblem::checkNonlocalCoupling()
```

Comment:
We could check first if _any_ of the existing kernels is nonlocal (not just the _active_ ones). Basically assemble a list of only the non local kernels and just check those if they are active during a timestep.